src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -6515,6 +6516,11 @@ internal void TdsLogin(SqlLogin rec, TdsEnums.FeatureExtension requestedFeatures
                 throw;
             }
 
+            if (outSSPIBuff != null)
```

Comment:
code above throws and in that case array would be not returned to pool, is it okay?

Reply:
It's safe in terms ot he array usage, When rented the renter has the only reference to the array so the array would revert to standard usage model and be cleaned up by the gc eventually. 

Since exceptional behavious is already occuring i don't know what's safe to do and any idea of performance improvements is nullified. I could try to return it to the pool anyway but is it of value to do so?